### ISSUE TITLE
GaussAdjoint support for DDEProblem

### ISSUE THREAD

**Issue by m-bossart** [SciML/SciMLSensitivity.jl#1074]
**Is your feature request related to a problem? Please describe.**

I'm attempting to perform sensitivity analysis for a large (100s-1000s parameters+states) DDEProblem. Based on the recommendations, GaussAdjoint seems to be the best option, however the current implementation only supports ODEProblems per the documentation.  

**Describe the solution you’d like**

Support for DDEProblem + GaussAdjoint sensitivity analysis.

**Comment by devmotion** [comment]
Ref https://github.com/SciML/SciMLSensitivity.jl/issues/281

AFAIK there's no support for DDEs in SciMLSensitivity yet.

**Comment by m-bossart** [comment]
Thanks for linking that issue. There is an [example ](https://docs.sciml.ai/SciMLSensitivity/stable/examples/dde/delay_diffeq/) and a [test ](https://github.com/SciML/SciMLSensitivity.jl/blob/master/test/layers_dde.jl) along with other mentions of compatibility for DDEs so I figured there was some support?

**Comment by ChrisRackauckas** [comment]
There's direct differentiation but no adjoint yet.

**Comment by m-bossart** [comment]
Ok. And just to clarify `ReverseDiffAdjoint()` is considered direct differentiation not adjoint?

**Comment by ChrisRackauckas** [comment]
yes, it's a discrete adjoint not continuous. I meant no continuous adjoint.
